### PR TITLE
*: Remove promsum datasources, and ReportPrometheusQueries and replace with prometheusMetricsImporter datasources

### PR DESCRIPTION
**Pull request by chancez**
Removes ReportPrometheusQuery resources and renames the `spec.promsum` section
to `spec.prometheusMetricsImporter` for ReportDataSource's.

This moves all reportPrometheusQuery's promQL expressions into each ReportDataSource's `spec.prometheusMetricsImporter.query` field.

### CODE REVIEW

File: Documentation/writing-custom-queries.md [hunk 1]
```diff
@@ -3,10 +3,11 @@
 One of the main goals of Operator Metering is to be flexible, and extensible.
 The way this has been done is to use Kubernetes Custom Resources as a way of letting users add to, or expand upon the built-in reports and metrics that the operator already has.
 
-The primary Custom Resources that allow this are the [ReportPrometheusQuery][reportprometheusqueries], [ReportDataSource][reportdatasources], and the [ReportGenerationQuery][reportgenerationqueries].
+The primary Custom Resources that allow this are the ReportDataSource][reportdatasources], and the [ReportGenerationQuery][reportgenerationqueries].
```
Comment (timflannagan1) [inline Documentation/writing-custom-queries.md]: Missing the first open bracket in `ReportDataSource]`

Comment (chancez) [reply Documentation/writing-custom-queries.md]: Fixed

**Pull request merged**